plugin-css-modules.js

```diff
@@ -0,0 +1,139 @@
+/*
+ *
+ * Manages web standard resource related operations for CSS.
+ * This is a Greenwood default plugin.
+ *
+ */
+import fs from 'fs';
+import htmlparser from 'node-html-parser';
+import { parse, walk } from 'css-tree';
+import { ResourceInterface } from '@greenwood/cli/src/lib/resource-interface.js';
+import * as acornWalk from 'acorn-walk';
+import * as acorn from 'acorn';
+
+const cssModulesMap = new Map();
+
+function walkAllImportsForCssModules(scriptUrl, sheets, scope) {
+  const scriptContents = fs.readFileSync(scriptUrl, 'utf-8');
+
+  acornWalk.simple(acorn.parse(scriptContents, {
+    ecmaVersion: '2020',
+    sourceType: 'module'
+  }), {
+    ImportDeclaration(node) {
+      const { specifiers = [], source = {} } = node;
+      const { value = '' } = source;
+
+      // TODO bare specifiers support?
+      if(value.endsWith('.module.css') && specifiers.length === 1 && specifiers[0].local.name === 'styles') {
+        console.log('WE GOT A WINNER!!!', value);
+        const cssModuleUrl = new URL(value, scriptUrl);
+        const scope = cssModuleUrl.pathname.split('/').pop().split('.')[0];
+        const cssContents = fs.readFileSync(cssModuleUrl, 'utf-8');
+        const classNameMap = {};
+        let scopedCssContents = cssContents;
+
+        const ast = parse(cssContents, {
+          onParseError(error) {
+            console.log(error.formattedMessage);
+          }
+        });
+
+        walk(ast, {
+          enter: function (node) {
+            if(node.type === 'ClassSelector') {
+              const { name } = node;
+              const scopedClassName = `${scope}-${name}`;
+              classNameMap[name] = scopedClassName;
+
+              scopedCssContents = scopedCssContents.replace(`\.${name}`, `\.${scopedClassName}`);
+            }
+          }
+        });
+
+        // TODO could we convert this module into an instance of CSSStylesheet to grab values?
+        // https://web.dev/articles/constructable-stylesheets
+        // or just use postcss-modules plugin?
+        cssModulesMap.set(cssModuleUrl.pathname, {
+          module: classNameMap,
+          contents: scopedCssContents
+        })
+        // sheets.push(cssContents);
+      } else if(node.source.value.endsWith('.js')) {
```

Comment:
TODO - how to handle other file types / extensions?  Do we even need to check, can was assume something should already be there to handle it?